File: javascript/anagramChecker.js
function isAnagram(str1, str2) {
    // Remove spaces and convert to lowercase
    const normalize = str => str.replace(/\s+/g, '').toLowerCase();

    str1 = normalize(str1);
    str2 = normalize(str2);

    if (str1.length !== str2.length) return false;

    // Sort and compare
    return str1.split('').sort().join('') === str2.split('').sort().join('');
}

// Example usage
const s1 = "Listen";
const s2 = "Silent";

if (isAnagram(s1, s2)) {
    console.log(`${s1} and ${s2} are anagrams.`);
} else {
    console.log(`${s1} and ${s2} are NOT anagrams.`);
} 

File: Java/AnagramChecker.java
import java.util.Arrays;

public class AnagramChecker {

    public static boolean isAnagram(String s1, String s2) {
        // Remove spaces and convert to lowercase for case-insensitive comparison
        s1 = s1.replaceAll("\\s", "").toLowerCase();
        s2 = s2.replaceAll("\\s", "").toLowerCase();

        if (s1.length() != s2.length()) {
            return false;
        }

        // Convert strings to char arrays
        char[] arr1 = s1.toCharArray();
        char[] arr2 = s2.toCharArray();

        // Sort arrays
        Arrays.sort(arr1);
        Arrays.sort(arr2);

        // Check if sorted arrays are equal
        return Arrays.equals(arr1, arr2);
    }

    public static void main(String[] args) {
        String str1 = "Listen";
        String str2 = "Silent";

        if (isAnagram(str1, str2)) {
            System.out.println(str1 + " and " + str2 + " are anagrams.");
        } else {
            System.out.println(str1 + " and " + str2 + " are NOT anagrams.");
        }
    }
}
